test/e2e/plugin_test.go

```diff
@@ -0,0 +1,175 @@
+}
+
+func (suite *PluginTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *PluginTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+func (suite *PluginTestSuite) TestPlugin() {
+	pluginImage := os.Getenv("DEMO_STORAGE_PLUGIN_IMAGE")
+	require.NotEmpty(t, pluginImage, "DEMO_STORAGE_PLUGIN_IMAGE must be set")
+
+	// create jaeger custom resource
+	name := "my-jaeger"
+	exampleJaeger := getJaegerPluginDefinition(namespace, name, pluginImage)
+
+	secret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "plugin-config",
+		},
+		StringData: map[string]string{
+			"config.json": "{}",
+		},
+	}
+	err := fw.Client.Create(goctx.TODO(), secret, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying secret")
+
+	log.Infof("passing %v", exampleJaeger)
+	err = fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+	defer undeployJaegerInstance(exampleJaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, name, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for deployment")
+
+	ports := []string{"0:16686"}
+	portForward, closeChan := CreatePortForward(namespace, name, "all-in-one", ports, fw.KubeConfig)
+	defer portForward.Close()
+	defer close(closeChan)
+	forwardedPorts, err := portForward.GetPorts()
+	require.NoError(t, err)
+
+	// Check that the plugins hard coded services response is there
+	url := fmt.Sprintf("http://localhost:%d/api/services", forwardedPorts[0].Local)
+	err = WaitAndPollForHTTPResponse(url, func(response *http.Response) (bool, error) {
+		body, err := ioutil.ReadAll(response.Body)
+		require.NoError(t, err)
+
+		resp := &services{}
+		err = json.Unmarshal(body, &resp)
+		if err != nil {
+			return false, nil
+		}
+
+		for _, v := range resp.Data {
+			if v == "dummy1" { // hardcoded service
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+	require.NoError(t, err, "Failed waiting for expected content")
+}
+
+func getJaegerPluginDefinition(namespace, name, pluginImage string) *v1.Jaeger {
+	exampleJaeger := &v1.Jaeger{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1.JaegerSpec{
+			Strategy: v1.DeploymentStrategyAllInOne,
+			Storage: v1.JaegerStorageSpec{
+				Type: "grpc-plugin",
+				GRPCPlugin: v1.GRPCStoragePluginSpec{
+					Image:             pluginImage,
+					Binary:            "/plugin/demo-storage-plugin",
+					ConfigurationFile: "/etc/plugin-config/config.json",
+				},
+			},
+			JaegerCommonSpec: v1.JaegerCommonSpec{
```

Comment:
Is this required? Sounds like it's mostly plumbing that the operator could take care of?

Reply:
I did not think of this before, but ee can also have SecretName as part of the option and generate this code.

Either way, plugins only support one file, so we would then have to make a decision between Secret or ConfigMap here..

Reply:
The config is fine, I was thinking about the `plugin-volume`.